feat: Planar Clarity Video Wall

GabFitzgerald opened the pull request:
First attempt
Resolves #59 

some confusion with options / **options

KimBurgess commented:
Having a quick read now. To provide some further reading on `**options`, that's a use of [splatting](https://crystal-lang.org/reference/syntax_and_semantics/splats_and_tuples.html).

TL;DR it can be used as a continent way to forward an arbitrary set of keyword arguments around. It was used a lot in the previous Ruby drivers, but would recommend avoiding it unless absolutely needed here as it can lead to some sloppy type signatures.

KimBurgess commented on drivers/planar/clarity_matrix.cr:
```diff
@@ -0,0 +1,111 @@
+    transport.tokenizer = Tokenizer.new("\r")
+  end
+
+  def on_unload
```
When writing drivers, if there aren't any actions that you need to run in these lifecycle callback methods, you can omit them.

KimBurgess commented on drivers/planar/clarity_matrix.cr:
```diff
@@ -0,0 +1,111 @@
+
+  def connected
+    schedule.every(60.seconds) { do_poll }
+    do_poll
```
`schedule.every` takes an optional second arg. It this is `true` it will run the scheduled block straight away in addition to the schedule so that you don't need to repeat like this.

KimBurgess commented on drivers/planar/clarity_matrix.cr:
```diff
@@ -0,0 +1,111 @@
+  # def power(state, broadcast_ip = false, options = {})
+  # what does broadcast_ip do here??
+  def power(state : Bool = false)
+    puts "power method running" 
```
Rather than using ours, take a look at the [logger](https://github.com/PlaceOS/drivers/blob/master/docs/writing-a-driver.md#logger).

KimBurgess commented on drivers/planar/clarity_matrix.cr:
```diff
@@ -0,0 +1,111 @@
+    # send("op A1 display.power = off \r")
+    if state == true
+      send("op A1 display.power = on \r") # changed ** -> A1: "Power should only be sent to display A1"
+      power?
```
It looks like the power state gets updated in remove to the power on command above, this second query can be removed so there's not unnecessary comms.

KimBurgess commented on drivers/planar/clarity_matrix_spec.cr:
```diff
@@ -0,0 +1,10 @@
+DriverSpecs.mock_driver "Planar::ClarityMatrix" do
+  # on connect it should do_poll the device
+  should_send("op A1 display.power ? \r")
```
When writing specs, you also want to mock mock the device responses so that you test full request -> response behaviour. For some info on this, see: https://github.com/PlaceOS/drivers/blob/master/docs/writing-a-spec.md#testing-streaming-io.

KimBurgess commented on drivers/planar/codes.txt:
```diff
@@ -0,0 +1,17 @@
+op ** display.power = on             // Power ON **check this
```
Try to keep any notes like this, or other items not specifically relating to the driver out of this repo.

dukeraphaelng commented:
> Having a quick read now. To provide some further reading on `**options`, that's a use of [splatting](https://crystal-lang.org/reference/syntax_and_semantics/splats_and_tuples.html).
> 
> TL;DR it can be used as a continent way to forward an arbitrary set of keyword arguments around. It was used a lot in the previous Ruby drivers, but would recommend avoiding it unless absolutely needed here as it can lead to some sloppy type signatures.

Hi Kim, upon going through this comment I thought it'd be a good idea to implement `Hash` instead of `splat`. However, this is not allowed since `splatting` in drivers is mostly used in 

```
queue(**({
      name: command,
    }.merge(options))) do
...
end
```

and queue's method is defined in `placeos-driver/src/placeos-driver.cr` as follows (line 114-117)
```
  # Queuing
  def queue(**opts, &block : Task -> Nil)
    @__queue__.add(**opts, &block)
  end
```
which disallows anything but splats and NamedTuple in the argument.

pkheav reviewed (changes requested):
I've had a look through and have picked up some things on style/feedback that @KimBurgess has given me previously. There are probably other things I have missed or not knowledgable enough to give feedback on so best to reach out to @KimBurgess for further feedback but this should hopefully be a good start of some things to look out for :)

stakach committed:
fix(cisco dna spaces): activation response object format

stakach committed:
fix(cisco dna spaces): start streams once keys obtained

stakach committed:
chore(shard.lock): bump versions

improved proxy support

w-le committed:
feat(mulesoft/bookings): query the current MONTH instead of DAY, as it makes the upstream api respond much faster for some reason

stakach committed:
fix(cisco dna spaces): api_key use before setting

stakach committed:
chore(shards.lock): bump versions

introduces more flexible proxy configuration

stakach committed:
feat(cisco dna spaces): add monitoring api calls

to provide DNA spaces dashboard with status of the streams being consumed

stakach committed:
feat: add whispir sms messaging service driver

stakach committed:
feat: add support for mapping groups based on a name prefix

stakach committed:
chore(cisco dna spaces): crystal tool format

stakach committed:
chore(cisco dna spaces): allow status updates to be pushed manuallly

by an admin, mostly for testing

stakach committed:
fix(cisco dna spaces): status updates not exposed to public interface

stakach committed:
fix(floorsense desks): parsing wrong model for floors request

fixes #107

stakach committed:
chore: update to latest version of action controller

w-le committed:
fix(mulesoft/bookings): clear schedule when changing polling period

stakach committed:
feat(cisco meraki): add user agent string for reporting

stakach committed:
feat(location services): add mac address mapping webhook

to support user login scripts

stakach committed:
fix(place location service): check webhook input

stakach committed:
feat(place staff api): add guest details and zones query

pkheav committed:
Sharp display (#99)

* feat(sharp_pn): initial commit

* feat(sharp_pn): port over rest of driver

* feat(sharp_pn): add basic spec

* feat(sharp_pn): add spec for do_poll

* feat(sharp_pn): add delay/timeout for commands

* fix(sharp_pn): remove debugging

* chore(sharp_pn): crystal tool format

pkheav committed:
Lg lcd (#61)

* feat(lg): initial commit

* feat(lg): add more methods

* feat(lg): add delay to some commands

* feat(lg): add cases on data receive

* fix(lg): handle commands that use the same value

* fix(lg): use safe self getter

* feat(lg): add specs for connected/do_poll

* feat(lg): add specs for power on/off

* fix(lg): use string form for query in specs

* feat(lg): group similar response behaviour

* chore(nec): add comments for Command1

* feat(lg): add test for input switching

* chore(lg): use warn instead of debug level logging

Co-authored-by: Kim Burgess <kim@place.technology>

* fix(lg): clear queue after hard_off

Co-authored-by: Kim Burgess <kim@place.technology>

* refactor(lg): use send and callback

* fix(lg): move connected to instance var

* fix(lg): update connected state

* chore(lg): run crystal format

* feat(lg ls5): implement minimal mute interface requirement

Co-authored-by: Philip Kheav <pkheav@Philips-MacBook-Pro.local>
Co-authored-by: Kim Burgess <kim@place.technology>
Co-authored-by: Stephen von Takach <steve@vontaka.ch>

dukenguyenxyz committed:
Epson projector (#62)

* wip: epson projector conversion from ruby

* --no-edit

* wip

* wip

* docker-compose up --build not working

* wip

* fix(epson_proj): make compilable

* fix(epson_proj): use MuteLayer enum

* feat(epson_proj): add working send

* fix(epson_proj): use Array to store errors

* feat(epson_proj): add specs for do_poll

* feat(epson_proj): add more specs

* fix(epson_proj): store commands in hash

* fix(epson_proj): move unmute_volume to instance var

* feat(epson projector): simplify mute interface

crystal tool format

Co-authored-by: PlaceOS <support@place.technology>
Co-authored-by: Philip Kheav <philip.kheav@hotmail.com>
Co-authored-by: Stephen von Takach <steve@vontaka.ch>

pkheav committed:
Panasonic display (#85)

* feat(pan_display): initial commit

* fix(pan_display): use specific specs for display

* fix(pan_display): remove projector specific code

* chore(pan/display): remove TODOs

pkheav committed:
Nec projector (#92)

* feat(nec_proj): initial copy from display driver

* feat(nec_proj): add command hash

* feat(nec_proj): add more functions

* feat(nec_proj): add more methods

* feat(nec_proj): add part of received method

* feat(nec_proj): add access for request in received

* feat(nec_proj): add more response processing

* chore(nec_proj): cleanup with crystal tool format

* chore(nec_proj): remove not needed commas

* feat(nec_proj): add input mapping for responses

* feat(nec_proj): add more response handling

* feat(nec_proj): add power status response handling

* fix(nec_proj): correct checksum calculation

* feat(nec_proj): add first passing spec

* feat(nec_proj): add spec for input?

* feat(nec_proj): add specs for do_poll

* feat(nec_proj): add proper specs for lamp/filter usage

* feat(nec_proj): store responses in enum

* chore(nec_proj): run crystal tool format

* chore(nec_port): use from_bytes? instead

* feat(nec_proj): add volume response processing

* feat(nec_proj): add spec for switch_audio

* feat(nec_proj): add more mute response handling

* chore(nec_proj): remove debugging

* chore(nec_proj): add type for command

* chore(nec_proj): make comment more clear

* feat(nec np series): simplify mute implementation

Co-authored-by: Stephen von Takach <steve@vontaka.ch>

pkheav committed:
feat(mdc): set targets to nil once stable (#74)

* feat(mdc): set targets to nil once stable

* fix(mdc): remove stable instance vars

pkheav committed:
Fixes for place/bookings.cr for o365 (#67)

* fix(bookings): pass correct params

* fix(bookings): use PlaceCalendar::Event::Attendee

* fix(bookings): remove default invalid calendar_id

* fix(bookings): set default calendar_id to nil

* fix(bookings): calculate ending using starting

* fix(bookings): default owner to @calendar_id

* fix(bookings): pass nil for owner

* fix(bookings): update booking info after book_now

* fix(bookings): set correct attendee fields for room

* fix(bookings): correct typo for status

* fix(bookings): remove o365 specific attendee status

* fix(bookings): set room response to accepted

* fix(bookings): allow owner of booking to be specified

* fix(bookings): always use system/room email

* fix(bookings): add non nil assertion for room_email

pkheav committed:
Global cache (#76)

* feat(gc): initial commit

* fix(gc): get code to compile

* fix(gc): remove delimiter

* fix(gc): split relay and port config

* feat(gc): add more specs

* feat(gc): add specs for all methods

* fix(global cache gc100 spec): requires delay before testing values

Co-authored-by: Stephen von Takach <steve@vontaka.ch>

pkheav committed:
QSC Audio DSP External Control (#96)

* chore(qsc): initial commit

* feat(qsc): get something compiling

* feat(qsc): add mute/fader functions

* feat(qsc): add helper to update change group

* feat(qsc): add quick dirty port of shellsplit

* feat(qsc): expose request in response handling

* feat(qsc): port over rest of driver

* chore(qsc): run crystal tool format

* feat(qsc): add basic specs

* fix(qsc): move login to on_update

* fix(qsc): remove wait: false from options

* feat(qsc): add specs for emergency id

* fix(qsc): change id to String instead of Int

* feat(qsc): add specs for mute/faders

* feat(qsc): add specs for phone methods

* fix(qsc): reduce spec to bare basic

* chore(qsc): check for emergency setting inline

* Revert "chore(qsc): check for emergency setting inline"

This reverts commit 6aaea447e86a796dc4bdfeaaa9b44b975a7cfd1f.

* fix(qsc): uncomment lines

pkheav committed:
Qsc audio dsp (#98)

* feat(qsc): initial commit

* feat(qsc): add more functions

* feat(qsc): more methods

* feat(qsc): allow single or multiple outputs

* feat(qsc): port over more stuff

* fix(qsc): use correct set for Types

* feat(qsc): allow faders to handle Bool for mute

* chore(qsc): rename alias Val to Num for clarity

* fix(qsc): correct Numue to Value

* feat(qsc): port response handling

* feat(qsc): add specs and login if needed

* fix(qsc): add proper JSON::Any casting

* feat(qsc): add more specs

* feat(qsc): add spec for component_get

* fix(qsc): store both pos and val

* chore(qsc): crystal tool format

* fix(qsc): use correct param types

* chore(qsc): cleanup todos

* chore(qsc): remove TODOs

* chore(qsc): remove more lines

w-le committed:
chore(mulesoft/bookings/spec): new response format

w-le committed:
feat(mulesoft/bookings): shorten query window from current month to 2 week sliding window

stakach committed:
fix(cisco dna spaces): monitoring API should use PUT

w-le committed:
feat(mulesoft/timetable): add unit of study code

w-le committed:
Merge branch 'master' of github.com:PlaceOS/drivers

w-le committed:
fix(mulesoft/bookings): unitCode may be Null

w-le committed:
fix(mulesoft/bookings): use event.recurring_master_id instead of unit_code, to align with existing frontend CalendarEvent model

w-le committed:
feat(mulesoft/bookings): event.title is now UnitName not activityName

w-le committed:
fix(mulesoft/bookings): model var assignemtn

w-le committed:
fic(mulesoft/bookings): event.title may be Null

stakach committed:
feat(floorsense): don't return unoccupied desk data

stakach committed:
feat(vergesense): only return data where people at the location

stakach committed:
feat(place desk bookings): include the booking type

stakach committed:
feat(xy sense loc service): don't return desk unless occupied

stakach committed:
feat(place area management): move level_id to be a tag

as this is how it is for other status items

stakach committed:
feat(floorsense): add booking requests

stakach committed:
fix(floorsense models): user doesn't need to have an email

stakach committed:
fix(meraki dashboard): no longer treat times as relative

this was a workaround for an issue where old locations were being returned

stakach committed:
feat(floorsense): add support for booking information

stakach committed:
chore: crystal tool format fixes

stakach committed:
Freespace driver (#119)

* feat: initial work on freespace driver

* feat(freespace): complete location services

* feat(freespace): use flatmap over map + flatten

Co-authored-by: Caspian Baska <caspian@place.tech>

Co-authored-by: Caspian Baska <caspian@place.tech>

stakach committed:
fix(floorsense desk): at_location expects an array

related to #120 might not solve the issue as the docs were not detailed

stakach committed:
chore(docs writing a driver): add pre_processor, cross system comms

and defining settings

stakach committed:
feat(cisco meraki): return the nearest WAP if there are no observations

stakach committed:
chore(freespace sensors): crystal tool format

stakach committed:
fix(cisco meraki): devices might not be assigned to a floor

stakach committed:
feat(cisco meraki): sync_floorplan_sizes to return floor plan info

useful while configuring networks

stakach committed:
feat(cisco meraki): add helpers for inspecting state

w-le committed:
feat(placeos/staff-api): add update_guest()

stakach committed:
fix(cisco meraki): add spec for calculating wap locations

stakach committed:
chore(shard.lock): bump versions

stakach committed:
chore(shards): bump driver version

stakach committed:
chore(shard.lock): bump driver version

w-le committed:
fix(placeos/staff-api): api endpoint of update guest

tob1k committed:
Exterity avedia player (#109)

* Initial exterity driver work

* progress

* Finalised 92xx/93xx drivers and specs

polonski committed:
Denon AV Receiver driver integration (#113)

* feat(denon):first cut, weird error

* feat(av_receiver):all specs passing

websymphony committed:
Biamp drivers (#110)

* Adds biamp drivers

* Rearrange the responds

stakach committed:
chore: crystal tool format

stakach committed:
feat(biamp): used named commands to speed execution

as you don't need to pass through all states if multiple changes are made to a particular fader as long as you end on the final state

MrYuion committed:
chore(spec-runner): update build to latest (#127)

stakach committed:
fix(cisco meraki): access points y location needs inverting

stakach committed:
Revert "fix(cisco meraki): access points y location needs inverting"

This reverts commit a360461a179afb3fbe85827426491386a0814428.

stakach committed:
bump to crystal 1.0.0 (#125)

* chore: bump to crystal 1.0.0

* feat: add driver spec report

so we can see what is failing looking at travis

* feat(report): improve report details

* fix(report): was outputting timeouts instead of no compile

* build: bump models

Co-authored-by: Caspian Baska <caspianbaska@gmail.com>

w-le committed:
feat(mulesoft/bookings): reduce query window to +/- 1d not 1w

stakach committed:
fix: LG and NEC driver compilation

upcase param must be explicit

kimburgess committed:
feat: Lenel OpenAccess (#24)

stakach committed:
chore(shard.lock): revert shard.lock to pre-v1 state (#133)

* chore(shard.lock): revert shard.lock to pre-v1 state

* chore(responsible): add shard to lockfile

* fix(shard.lock): add inactive support

stakach committed:
fix: revert shards to pre-v1

stakach committed:
feat(floorsense): add booking sync capabilities

stakach committed:
fix(floorsense booking sync): don't raise if floor not found

Commit:
build(docker-compose): remove the container_name key for redis (#134)

stakach committed:
feat(floorsense desks): add user and rfid operations

stakach committed:
feat(floorsense): add support for user syncing

stakach committed:
feat(floorsense): add a method for finding the user-id of a CSN

stakach committed:
fix(floorsense desks): expected event log ordering

stakach committed:
fix(floorsense desks): sort by time, then id

stakach committed:
fix(floorsense desk): add support for event logs after param

as using event id returns logs around the id versus strictly after it

stakach committed:
fix(floorsense sync): improved event log query

pkheav committed:
Sony projector (#136)

* feat(sony_serial): initial commit

* feat(sony_serial): port over rest of driver

* feat(sony proj serial): add basic spec

* feat(sony serial): add complete specs

* feat(sony fh): initial commit

* feat(sony fh): port over most of driver

* feat(sony fh): update response handling

* feat(sony fh): add specs

* feat(sony serial): add delimiter

* fix(sony_serial_proj): pass block to join

* chore(sony_proj): crystal tool format

* feat(sony fh): add switchable interface

* feat(sony serial): add switchable interface

Co-authored-by: tassja <xtassja@gmail.com>

stakach committed:
chore(floorsense sync): add some additional debugging

stakach committed:
fix(floorsense sync): staff api query needs to be sending unix timestamps

stakach committed:
fix(floorsense sync): compile error

tassja committed:
Sony LCD (#128)

stakach committed:
fix(floorsense sync): handle error where user wasn't in placeos

stakach committed:
fix(floorsense bookings sync): id 0 padding on correct side

stakach committed:
fix(floorsense desk): booking requires cid

plan_id doesn't work even though it should according to the documentation

stakach committed:
fix(floorsense sync): add a mutex around sync operations

stakach committed:
feat(place staff api): add delete booking to staff API

stakach committed:
fix(place staff api): get booking was using post instead of get

stakach committed:
fix(floorsense sync): ignore bookings that are not adhoc

when parsing the event log for booking creation events

stakach committed:
fix(floorsense sync): ignore bookings that are not adhoc

stakach committed:
fix(floorsense desks): set desk expiry time to ending time

stakach committed:
fix(place staff api): use patch for booking update

stakach committed:
fix(floorsense booking sync): book for now if start time in the past

stakach committed:
fix(floorsense sync): fix booking matching during sync

stakach committed:
fix(floorsense sync): release bookings that are not in PlaceOS

stakach committed:
fix(floorsense sync): don't sync bookings to foor that ended in the past

stakach committed:
fix(floorsense sync): check ending slightly in the future

as there might be issues with slight clock drift between servers

stakach committed:
feat(floorsense desks): check for errors in rfid create response

stakach committed:
fix(floorsense models): return true where there is no response data

Commit:
fix: extend NEC Display audio inputs (#137)

stakach committed:
feat(floorsense sync): ensure card numbers are always up to date

stakach committed:
fix(floorsense sync): ensure bookings are started in floorsense

as clock drift might mean they didn't start straight away

Commit:
fix(sony/projector): fix bug in rendering bytes (#138)

Adds named arguments to to_s invocation on the checksum. This driver previously failed to compile.

stakach committed:
fix(user group mapping): preference login_name over email

when querying for user information

stakach committed:
feat(docs emails): guide to writing a driver that sends emails

stakach committed:
feat(meraki dashboard): split into a service and a logic

w-le committed:
chore(shard.lock): bump calendar and office365 (#145)

fix event.sensitivity and fetch events limited to 10
https://github.com/PlaceOS/calendar/commit/ab8babafca01954545694da5cc116108a7af4eb3
https://github.com/PlaceOS/office365/pull/17

w-le committed:
fix(place/calendar): Dont add Teams link to every Event by default

w-le committed:
fix(place/calendar): o365 event timezones

w-le committed:
Revert "feat(mulesoft/bookings): reduce query window to +/- 1d not 1w"

This reverts commit 2db8ff583bb7a995ae0c8eec6c92ad02aa6568fa.

stakach committed:
feat(sony projector): add pj talk protocol (#141)

* feat(sony projector): add pj talk protocol

* Update drivers/sony/projector/pj_talk.cr

Co-authored-by: Kim Burgess <kim@place.technology>

* Update drivers/sony/projector/pj_talk.cr

Co-authored-by: Kim Burgess <kim@place.technology>

* minor tweaks and commet updates

Co-authored-by: Kim Burgess <kim@place.technology>

stakach committed:
chore(floorsense): add booking sync overview

Commit:
build: bump to placeos-driver 5.0.8 (#135)

Fixes an issue with symbol resolution in the spec runner

stakach committed:
feat: add generic MQTT driver (#148)

* feat: add generic MQTT driver

allows for a single connection to be used for multiple drivers
once configured, various drivers can subscribe to interested state via typical PlaceOS subscription model

* fix: driver mock description

Co-authored-by: Caspian Baska <caspian@place.tech>

stakach committed:
chore(place mqtt): add some additional logging

stakach committed:
feat(place mqtt): add ping to keep session alive

stakach committed:
chore(place mqt): log when a ping is being sent

stakach committed:
chore(shard.lock): bump driver version

stakach committed:
chore(shard.lock): bump driver version

w-le committed:
feat(mulesoft/bookings): polling interval is cron not minutes

Commit:
ci(github/style): update style action

w-le committed:
feat(place/calendar): support event.location

stakach committed:
fix(meraki locations): skip and warn about invalid locations

fixes #150

stakach committed:
fix(meraki locations): check for invalid locations

Commit:
style: define base ameba config and fix lints across drivers (#149)

* build: basic ameba config

* style(cisco/dna_spaces): fix lints

* style(extron/matrix): fix lints

* style(lenel): fix lints

* style(mulesoft/booking_api): fix lints

* style(meraki): fix lints

* style(xovis): fix lints

* style(floorsense): fix lints

* style(cisco/switch/snooping_catalyst): fix lints

* style(place/calendar): fix lints

* style(xy_sense/location_service): fix lints

* style(vergesense/vergesense_api): fix lints

* style(denon/amplifier/av_receiver): fix lints

* style(microsoft/find_me_spec): fix lints

* style(screen_technics/connect): fix lints

* build(lenel/open_access): include responsible

kimburgess committed:
fix(lenel): build regression (#152)

* fix: missing inactive-support dependency

* fix: incorrect type definitions

Commit:
feat(global_cache): conform to electrical relay interface (#154)

stakach committed:
feat: add echo360 driver (#155)

stakach committed:
fix(echo360): incorrect path generation in macro

stakach committed:
feat(echo360): add some debugging output

stakach committed:
fix(echo360): next can be an empty string

pkheav committed:
feat(smtp): add setting to ignore cert validation (#161)

* feat(smtp): add setting to ignore cert validation

* fix(smtp): add check for setting

to ignore ssl verify

Commit:
ci: move to github  (#153)

* build(shard.lock): bump driver and exec_from
* build(Dockerfile): improve layer caching
* ci(github): add github ci
* ci(github): use `docker exec` instead of `docker-compose exec`
* ci(docker-compose): mount .git for purpose of server specs

w-le committed:
feat(mulesoft): add Calendar exporter

stakach committed:
fix: clear schedules on connect (#170)

* fix: clear schedules on connect

to test if disconnect is not being called

* fix(nexia): clear schedule on connect

Co-authored-by: Viv Briffa <viv@place.technology>

Commit:
lint(mulesoft): fix ameba lints (#171)

stakach committed:
chore(shard.lock): bump driver and tasker versions

stakach committed:
chore(shard.lock): bump versions

stakach committed:
feat(cisco dna spaces): add BLE RSSI update message support

fixes #162

pkheav committed:
Cisco ise (#122)

* feat(cisco ise): add initial basic guest driver

* feat(ise spec): add beginning of ise spec and cleanup ise driver

* fix(ise): ensure request body is not IO | nil before parsing in ise spec

* fix(ise): initialise @portal_id

* chore(ise): use instance variables instead

* chore(ise spec): fix typo

* fix(guests): use uri_base

* feat(ise): add more settings

* feat(ise): decode xml

* feat(ise): add specs for xml fields

* chore(ise): update default settings

* feat(ise): add params to override default values

* fix(ise): deal with single names

* feat(ise): add mandatory headers

* feat(ise): add basic error handling

* feat(ise): use UUID to generate usernames

* feat(ise): add support for optional sms details

* chore(ise): crystal tool format

* fix(ise): simplify request body check

* fix(ise): make more generic

* chore(ise): use indentation to make xml clearer

* feat(ise): support timezone setting

* fix(ise): convert to local time properly

* fix(ise): correct spec structure

* feat(ise): add support to get guest credentials

* feat(ise): add support for mock data

* fix(ise): generate basic auth token from creds

* feat(ise): add link v2.2 api docs

* fix(ise): make compatible with both v1.4/v2.2

* fix(ise): update XML to match schema

* chore(ise): crystal tool format

* feat(ise): add support to return custom data

* feat(ise): use ISE generated credentials

* fix(ise): make mock data similar to real api

* feat(ise): generate more consistent usernames

Co-authored-by: Cameron Reeves <cam@Camerons-MacBook-Pro.local>

pkheav committed:
chore(ise): run crystal tool format (#174)

stakach committed:
feat(cisco dna spaces): add support for IoT telemetry

stakach committed:
feat(place calendar): enable proxy configuration from backoffice

stakach committed:
feat(cisco dna spaces): updates to IoT telemetry payload

stakach committed:
feat(placeos staff api): add checked in filter to bookings query

stakach committed:
chore(place staff-api): fix checked_in param

stakach committed:
feat(floorsense desks): add websocket version

stakach committed:
feat(floorsense sync): update for use with websocket

stakach committed:
fix(floorsense desk websocket): add additional debugging

stakach committed:
chore(shard.yml): bump placeos driver version

stakach committed:
feat(floorsense): implement booking confirmation

so bookings can be confirmed by checking-in placeos bookings

stakach committed:
chore(shards.lock): bump driver version

stakach committed:
fix(docker-compose): mount www folder

stakach committed:
chore(shard.lock): bump driver version

adds support for websocket proxies

stakach committed:
chore(shard.lock): bump connect proxy version

stakach committed:
chore(floorsense docs): include latest updates

in the documenation

stakach committed:
feat(AWS SNS): add SNS SMS service

w-le committed:
fix(placeos/area_management): use duplication_factor in area counts. so the estimates returned are of people not devices

stakach committed:
fix(aws sns sms): sign request

w-le committed:
fix(mulesoft/bookings): query_bookings is private

stakach committed:
chore(shard.lock): bump driver version

stakach committed:
chore(shard.lock): bump drivers version

stakach committed:
feat(placeos staff-api): add additional queries

for guests and events

stakach committed:
core(shard.lock): bump versions

tassja committed:
feat(spec-runner): seperate driver spec runner (#179)

* refactor: move test runner out of drivers repo

* fix: rename src to app

* refactor: remove local image context

* chore: update docker image

* chore: update docker-compose

* chore: restore files

* fix: restore report

* ci: update docker-compose

* chore: add report executable

* ci: add image to docker-compose

* fix: exit_code

* style: remove quotes of command

Just for consistency

* refactor: run shards check before install

Co-authored-by: Caspian Baska <caspian@place.tech>
Co-authored-by: Caspian Baska <email@caspian.computer>

stakach committed:
chore(shard.lock): bump versions

w-le committed:
chore(shard.lock): bump version of placeos-driver to 5.3.12

This adds a new feature enabling a Setting `host_header` to set a Host header to Service driver http requests

kimburgess committed:
fix(nexia): re-implement driver (#183)

+ seperates NTP protocol from device interactions
+ fix missing device ID in requests
+ neaten up response parsing
+ (breaking) drop methods that internally loop

stakach committed:
chore(shard.lock): bump driver version

Commit:
feat: implement harness script to interface with the spec runner (#181)

* feat: implement harness script to interface with the spec runner

* fix(harness): unset exit_code

stakach committed:
chore(palce calendar): update to allow for longer wait times

stakach committed:
chore(shard.lock): bump versions

w-le committed:
feat(place/visitor_mailer): add host mailbox to QR data (#187)

stakach committed:
Gallagher rest api (#156)

* feat: add gallagher rest api

* feat: add cardholder creation

* feat: finish gallagher

* chore(gallagher): add link to documentation

kimburgess committed:
fix: broken require (#182)

Resolves a breaking API change / layout change in upstream dep.

Introduced in 872a61c3a846a2ec3887ec39cf8141a3eb6b2f24.

caspiano committed:
style: fix ameba lints

Commit:
chore(shard.lock): bump driver (#188)

stakach committed:
fix(palce staff-api): ensure host header set on authentication

host header override

kimburgess committed:
Extron (#142)

* test: unhandled device responses

* feat: debug messages for tx/rx

* fix: incorrect clock format

* fix: await unsolicited messages on connect

* fix: exhaustive case for unsolicted comms

* fix: specify switcher size from settings

Different device models appear to provide unique response to the device
info query. Not all allow for introspection of device IO. Manually
specify for now.

* fix: incorrect type for settings

* fix: incorrect syntax for switch command

* feat: provide parsed responses

* style: crystal formatter

* style: remove unused assigns (ameba)

* feat: group volume control

* feat: audio mute control

Commit:
test: parallel spec runner (#194)

* test: write compilation failures to a log

* test: remove latest

This is in favour of explicitly specifying versions we are testing
against.

* fix(workflows/ci.yml): move if to right place

* test(workflows/ci): run specs for subset of drivers, before whole suite

* fix(harness): forward arguments to `report`

* fix(ci): remove anchors

* fix(workflows/ci): changes action

* fix(harness): pass files to report

* fix(ci): add build.cr

* ci: add shards cache

* ci: continue on error

* test: forward CI to docker-compose

* ci: continue on nightly errors

* docs(README): add `harness` docs

* ci: only upload *.log

Commit:
docs(harness): better messages (#195)

stakach committed:
chore(shard.lock): bump driver version

tassja committed:
feat(lenel): refactor to use built in before_request callback

* fix: change conflicting method name

* fix: correct use of before_request

* feat(lenel): refactor to use built in before_request callback

* feat: migrate to latest version of driver

Co-authored-by: Stephen von Takach <steve@vontaka.ch>

stakach committed:
chore(shard.lock): bump version

stakach committed:
feat(build.cr): move to drivers folder

stakach committed:
chore: add a sym link to src folder

stakach committed:
feat: migrate to standalone drivers

no requirement for build.cr

stakach committed:
chore(spec): mark mock classes

stakach committed:
chore(docs): don't include mocks in crystal docs gen

stakach committed:
feat: migrate specs to using `require "placeos-driver/spec"`

stakach committed:
fix(cisco meraki): drivers compilation failure

due to recent updates to standalone applications

w-le committed:
feat(lenel/openaccess): add more funcs; make badge response fields op… (#202)

* feat(lenel/openaccess): add more funcs; make badge response fields optional

* style(lenel/openaccess): crystal tool format

* fix(lenel/openaccess): badgekey to String before delete

stakach committed:
feat(meraki locations): add min variance override

where meraki might be too confident

stakach committed:
feat(cisco meraki): add regex filter on OS when searching

stakach committed:
feat(lenel open access): apply recommended changes

thanks @caspiano

stakach committed:
chore(shard.lock): bump place calendar version

w-le committed:
fix(meraki/locations): also override minimum variance when new data is more accurate

stakach committed:
chore(shard.lock): bump driver version

Commit:
ci: fix exit code of `report` (#204)

* ci: fixup exit of jobs

* style(lenel/open_access): format

* ci: better job names

* docs: reorder README

* feat(pre-commit): add pre-commit checks

* ci: remove duplicate continue-on-error

* ci: add dependabot for github actions

* Revert "feat(pre-commit): add pre-commit checks"

This reverts commit 39c2bdf4d30a595feb1e7f7d87c8696af25f5bca.

* fix(harness#report): swallow install stdout, keep stderr

tassja committed:
fix(officernd): refactor request functions (#203)

* refactor: add responsible

* style: fix indents

* refactor: request functions

* revert: remove responsible

* style: remove comments

* fix: label headers arg

stakach committed:
feat(gallagher): fix version comparison

resolves place-technology/suncorp-ntt#2

stakach committed:
feat(gallagher): log outgoing requests to help with debugging

resolves place-technology/suncorp-ntt#20

stakach committed:
feat(place staff api): add systems and building details queries

stakach committed:
feat(place bookings): add locatable interface

stakach committed:
feat(place location services): add bookings discovery

where a person is in a meeting

stakach committed:
chore(shard.lock): bump driver version

stakach committed:
feat(place location service): simplify map-reduce

stakach committed:
feat(location services): add methods for querying sensor data

Commit:
ci: disable fail fast (#206)

stakach committed:
chore(shard.lock): bump driver version

stakach committed:
feat(place staff api): add metadata writing helper

stakach committed:
feat(place area management): add support for sensor summaries

stakach committed:
fix(place area management): sensor details may not exist for a level

stakach committed:
feat(steinel hpd2): implement sensor interface

stakach committed:
fix(place area management): location might not exist

stakach committed:
fix(palce staff api): write_metadata description is required

stakach committed:
feat(sensors): update to latest sensor interface

stakach committed:
fix(steinel hpd2): sensor search matching types

stakach committed:
feat(xovis sensor): implement sensor interface

stakach committed:
fix(xovis sensor): allow queries to be configurable

stakach committed:
fix(place area management): fix incorrect sensor metadata key

stakach committed:
fix(area management): sensors are stored on each levels zone

stakach committed:
fix(place area management): ensure sensor level is configured

w-le committed:
fix(vergesense): Type of Space.People.coordinates (#207)

* fix(vergesense): Type of Space.People.coordinates

* style(vergesense): whitespace

stakach committed:
fix(place area management): fix sensor status

w-le committed:
Vergesense: space_ref_id can be null (#208)

* fix(vergesense): Type of Space.People.coordinates

* style(vergesense): whitespace

* fix(vergesense): Null space_ref_id seen in the wild

stakach committed:
feat(place area management): unify metadata collection

stakach committed:
fix(place area management): ensure level data is synced

before requesting sensor data

stakach committed:
fix(place area management): generate area summaries if only sensors

was previously looking for location objects for map details

stakach committed:
feat(floorsense): add support for locker operations

stakach committed:
feat(floorsense): simplify locker control

use locker key as the primary method for accessing lockers

stakach committed:
feat(xy sense location service): allow for manual sync

stakach committed:
fix(xy sense location services): space names can be null

w-le committed:
fix(staff-api): query events must GET not PATCH

stakach committed:
fix(gallagher): some card type fields are not required

stakach committed:
chore(place area management): fix ameba warning

stakach committed:
feat(ashrae bacnet): initial work on generic BACnet driver

stakach committed:
feat(microsoft findme): add additional logging output

pkheav committed:
Svsi (#166)

* feat(svsi): port decoder driver

* feat(svsi): port encoder driver

* fix(svsi): minor changes

* feat(svsi): port n_series_switcher

* feat(svsi_virtual_switcher): attempt to port

* fix(svsi): use correct variable name

* fix(svsi_virtual_switcher): use blocks

* fix(svsi_n_series): downcase property

* feat(svsi_virtual_switcher): implement Switchable

* fix(svsi_virtual_switcher): make types clearer

* feat(svsi_virtual_switcher): implement switch_to

* fix(svsi): implement more interfaces

* fix(svsi_virtual_switcher): decoder switch_to

* fix(svsi_virtual_switcher): use switch_to

* fix(svsi_virtual_switcher): workaround for int key

* fix(svsi_virtual_switcher): use correct indexes

* fix(svsi_virtual_switcher): use Int32 for input

* chore(svsi): clearly define delimiter

Co-authored-by: Caspian Baska <caspian@place.tech>

* style(svsi): run formatter

* fix(svsi): use delimiter directly

* refactor(svsi:decoder): use mapped_enum for commands

* refactor(svsi): ensure exhaustive case

* fix(svsi): construct do_send args

* fix(svsi): fix do_send

Co-authored-by: Caspian Baska <caspian@place.tech>
Co-authored-by: Caspian Baska <email@caspian.computer>

caspiano committed:
Merge branch 'master' into planar-video-wall
